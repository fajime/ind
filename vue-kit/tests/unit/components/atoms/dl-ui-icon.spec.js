import { mount } from '@vue/test-utils';
import DlUiIcon from '../../../../src/components/atoms/dl-ui-icon/index.vue';

describe('dl-ui-icon', () => {

  const wrapper = mount(DlUiIcon);

  it('Component Name', () => expect(DlUiIcon.name).toBe('dl-ui-icon'));

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('property title an description ', () => {
    const wrapperTest = mount(DlUiIcon, {
      propsData : {
        title       : 'foo',
        description : 'bar'
      }
    });
    expect(wrapperTest.element.textContent.includes('foo')).toBeTruthy();
    expect(wrapperTest.element.textContent.includes('bar')).toBeTruthy();
  });

});
