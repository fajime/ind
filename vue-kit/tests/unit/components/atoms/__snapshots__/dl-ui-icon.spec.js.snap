// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dl-ui-icon renders correctly 1`] = `
<div
  class="dl-ui-icon"
>
  <svg>
    <!--v-if-->
    <!--v-if-->
    <use
      xlink:href="#dl-ids-icon-user-single-outlined"
    />
  </svg>
</div>
`;
